examples/min-platform/embedding/src/wasi.rs

```diff
@@ -0,0 +1,859 @@
+        Executor(
+            EXECUTOR
+                .0
+                .borrow_mut()
+                .as_ref()
+                .expect("Executor::current must be called within block_on")
+                .0
+                .clone(),
+        )
+    }
+    pub fn push_deadline(&mut self, due: u64, waker: Waker) {
+        self.0.borrow_mut().schedule.push((due, waker))
+    }
+}
+
+// Schedule, as provided by the Deadline future impls. Map of due times to
+// wakers.
+struct ExecutorInner {
+    schedule: Vec<(u64, Waker)>,
+}
+
+impl ExecutorInner {
+    // Get the earliest deadline currently waiting. None if there are no
+    // deadlines.
+    fn earliest_deadline(&self) -> Option<u64> {
+        self.schedule.iter().map(|(due, _)| due).min().copied()
+    }
+    // Return all wakers associated with deadlines before or equal to the
+    // current clock time. Removes the wakers and their deadline from the
+    // schedule.
+    fn ready_deadlines(&mut self, now: u64) -> Vec<Waker> {
+        let mut i = 0;
+        let mut wakers = Vec::new();
+        // This is basically https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extract_if,
+        // which is unstable
+        while i < self.schedule.len() {
+            if let Some((due, _)) = self.schedule.get(i) {
+                if *due <= now {
+                    let (_, waker) = self.schedule.remove(i);
+                    wakers.push(waker);
+                } else {
+                    i += 1;
+                }
+            } else {
+                break;
+            }
+        }
+        wakers
+    }
+}
+
+// Yanked from core::task::wake, which is unfortunately still unstable :/
+fn noop_waker() -> Waker {
+    use core::task::{RawWaker, RawWakerVTable};
+    const VTABLE: RawWakerVTable = RawWakerVTable::new(
+        // Cloning just returns a new no-op raw waker
+        |_| RAW,
+        // `wake` does nothing
+        |_| {},
+        // `wake_by_ref` does nothing
+        |_| {},
+        // Dropping does nothing as we don't allocate anything
+        |_| {},
+    );
+    const RAW: RawWaker = RawWaker::new(core::ptr::null(), &VTABLE);
+
+    unsafe { Waker::from_raw(RAW) }
+}
+
+fn block_on<R>(clock: Clock, f: impl Future<Output = Result<R>> + Send + 'static) -> Result<R> {
+    // Guard against nested invocations
+    if EXECUTOR.0.borrow_mut().is_some() {
+        panic!("cannot block_on while executor is running!")
+    }
+    let executor = Executor::new();
+    *EXECUTOR.0.borrow_mut() = Some(Executor(executor.0.clone()));
+
+    // No special waker needed for this executor.
+    let waker = noop_waker();
+    let mut cx = Context::from_waker(&waker);
+    let mut f = core::pin::pin!(f);
+
+    // Drive the Future to completion in the following loop
+    let r = 'outer: loop {
+        const POLLS_PER_CLOCK: usize = 200; // Arbitrary. Could be as little
+                                            // as 1. Doesn't hurt to kee[
```

Comment:
Incomplete/truncated sentence in comment?

Reply:
Thanks